### ISSUE TITLE
Can't read lyrics tag when description is set

### ISSUE THREAD

**Issue by g-fb** [Serial-ATA/lofty-rs#383]
### Reproducer

I tried this code:

```rust
use lofty::prelude::ItemKey;
use lofty::probe::Probe;
use lofty::file::TaggedFileExt;

fn main() {
    let tagged_file = Probe::open("/path/to/file.mp3")
        .expect("ERROR: Failed to open file")
        .read();
    let tagged_file = tagged_file.unwrap();
    let tags = tagged_file.tags();
    for tag in tags {
        let lyrics = tag.get_string(&ItemKey::Lyrics).unwrap_or_default();
        println!("{lyrics}");
    }
}
```

- open file in Kid3 (tested with [this file](https://pixabay.com/music/beats-no-copyright-for-her-chill-upbeat-summel-travel-vlog-and-ig-music-202298/))
- add a `+ Tag2`
- add lyrics tag
- set a random description
- set text to "the lyrics"
- save file

### Summary

Can't read lyrics tag when description is set.

### Expected behavior

`println!("{lyrics}");` prints "the lyrics"

**Comment by Serial-ATA** [comment]
In the `Id3v2Tag` -> `Tag` conversion, frames with descriptions are intentionally left behind, as `Tag` has no way of retaining that information:

https://github.com/Serial-ATA/lofty-rs/blob/f17bac49b3f3e7757784c36de8220d2581a19d7a/lofty/src/id3/v2/tag.rs#L1173-L1188

When dealing with tags using anything outside of basic text storage, it's best to use the concrete tags ([`Id3v2Tag`](https://docs.rs/lofty/latest/lofty/id3/v2/struct.Id3v2Tag.html)) in this case.

Hopefully in the future there will be a way to preserve more of this information in a generic way, but right now `Tag` is pretty limited.

**Comment by g-fb** [comment]
So I should be able to get the lyrics with a description if I use `Id3v2Tag`?

If yes, then I don't understand how. I tried like this:

```rust
let mut tagged_file = lofty::read_from_path(file.path.clone()).unwrap();
let tag = tagged_file.tag_mut(TagType::Id3v2);
match tag {
    Some(tag) => {
        let tag: Id3v2Tag = <Tag as Clone>::clone(&(*tag)).into();
        const LYRICS_ID: FrameId<'_> = FrameId::Valid(Cow::Borrowed("USLT"));
        let lyrics = tag.get_text(&LYRICS_ID);
        println!("file: {}\n{lyrics:#?}", file.path.clone());
    }
    None => ()
}
```
But I still can't get the ones with a description.

PS: I suck at rust.

**Comment by Serial-ATA** [comment]
> But I still can't get the ones with a description.

So a few things:

* `Into::<Id3v2Tag>::into(tag)` will only turn to the generic `Tag` items into an `Id3v2Tag`. As the USLT frame is held back in the initial `Id3v2Tag` -> `Tag` conversion, it will not exist during this second conversion. I plan on fixing this in #302.
* [`Id3v2Tag::get_text`](https://docs.rs/lofty/latest/lofty/id3/v2/struct.Id3v2Tag.html#method.get_text) only works for text frames (TALB, TIT2, etc..). I should improve the docs for that.

The solution by @Ferry-200 is correct. If you read an `MpegFile` directly, you get access to the `Id3v2Tag`. You're currently using `read_from_path` which will give you a `TaggedFile`, and thus you only get access to `Tag`.

**Comment by Serial-ATA** [comment]
Nice, I'll keep this pinned to remind me to make descriptions and languages available in `TagItem` somehow.

**Comment by yanquer** [comment]
but , how can t read from other-ext-file, such as Flac--Music-File ?

**Comment by Serial-ATA** [comment]
Yes, in ID3v2 you can have potentially many lyrics with different languages and descriptions. If you aren't working with ID3v2, though, then there should only ever be one.
